src/enum.cr

```diff
@@ -276,6 +276,21 @@ struct Enum
     value.hash
   end
 
+  def each
+    {% if @type.has_attribute?("Flags") %}
+      return if value == 0
+      {% for member in @type.constants %}
+        {% if member.stringify != "All" %}
+          if includes?({{@type}}::{{member}})
+            yield {{@type}}::{{member}}
+          end
+        {% end %}
+      {% end %}
+    {% else %}
+      {% raise "Can't iterate #{@type}: only Flags Enum can be iterated" %}
```

Comment:
This isn't `Enum.each` but `Enum#each`. It iterates the flags included in the enum value, not all possible enums constants. For example:

```crystal
@[Flags]
enum MyEnum
  One
  Two
  Three
end
MyEnum.flags(One, Two).each do |flag|
  # yields One then Two
end
```

I don't see any value in iterating the single value of a non flags enum; I see value in iterating an enum constants. I'lll open another PR with this commit and `Enum.each`.